synaptic_reconstruction/inference/util.py

```diff
@@ -1,9 +1,92 @@
+import os
 import time
-from typing import Dict
+from glob import glob
+from typing import Dict, Optional
+
+import bioimageio.core
+import imageio.v3 as imageio
 import numpy as np
 import torch
 import torch_em
+import xarray
+
+from elf.io import open_file
 from torch_em.util.prediction import predict_with_halo
+from tqdm import tqdm
+
+
+def get_prediction(
+    input_volume: np.ndarray,  # [z, y, x]
+    model_path: str,
+    tiling: Dict[str, Dict[str, int]],  # {"tile": {"z": int, ...}, "halo": {"z": int, ...}}
+    verbose: bool = True,
+    with_channels: bool = False,
+):
+    """
+    Run prediction on a given volume.
+
+    This function will automatically choose the correct prediction implementation,
+    depending on the model type.
+
+    Args:
+        input_volume: The input volume to predict on.
+        model_path: The path to the model checkpoint.
+        tiling: The tiling configuration for the prediction.
+        verbose: Whether to print timing information.
+        with_channels: Whether to predict with channels.
+
+    Returns:
+        The predicted volume.
+    """
+    is_bioimageio = model_path.endswith(".zip")
+
+    # We standardize the data for the whole volume beforehand.
+    # If we have channels then the standardization is done independently per channel.
```

Comment:
I have moved the normalization logic here, so that we don't need this in the individual scripts.